aten/src/ATen/detail/AcceleratorHooksInterface.h

```diff
@@ -15,7 +15,32 @@ struct TORCH_API AcceleratorHooksInterface {
   virtual ~AcceleratorHooksInterface() = default;
 
   // Whether the device at device_index is fully initialized or not.
-  virtual bool hasPrimaryContext(DeviceIndex device_index) const = 0;
+  virtual bool hasPrimaryContext(DeviceIndex device_index) const {
+    return false;
```

Comment:
Why do you add an implementation here instead of keeping it pure virtual?

Reply:
Let me revert it back. It should be pure virtual if every device backend requires to implement it.